mars/learn/contrib/pytorch/dataset.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 1999-2020 Alibaba Group Holding Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+try:
+    import torch
+    from torch.utils.data import Dataset
+except ImportError:
+    torch = None
+    Dataset = object
+
+from ....core.context import get_context
+from ....tensor.core import TENSOR_TYPE
+from ....dataframe.core import DATAFRAME_TYPE, SERIES_TYPE
+from ....utils import require_not_none
+
+
+@require_not_none(torch)
+class MarsDataset(Dataset):
+    r"""MarsDataset that inherit from torch.utils.data.Dataset.
+    It converts from Mars basic datatype such as Tensor,
+    DataFrame, Series. Additionally, it's constructor can receive
+    np.ndarray, pd.DataFrame, pd.Series type.
+    """
+    def __init__(self, *tileables):
+
+        self._context = get_context()
+        self._tileables = tileables
+
+    def __len__(self):
+        return self._tileables[0].shape[0]
+
+    def __getitem__(self, index):
+        return tuple(self.get_data(t, index) for t in self._tileables)
+
+    @staticmethod
+    def get_data(t, index):
+        if isinstance(t, TENSOR_TYPE):
```

Comment:
If encounter Mars object, shall us `execute` it for the first time. Then, we can do index fetch only.

This is based on the fact that.

```python
df.execute()
df.iloc[:5].fetch()  # it's allowed to fetch subset of executed objects, but it would be very quick.
df.iloc[5:10].fetch()
```

Reply:
That is great advice. I have considered this situation, but forget to optimize it. I will modify it as soon as possible.